File: linked lists/19. Remove Nth Node From End of List/removeNthFromEnd.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    let len = 0;
    let node = head;
    
    while (node) {
        node = node.next;
        len++;
    }
    
    if (len === n) {
        return head.next;
    }
    
    node = head;
    for (let i = 0; i < len - 1 - n; i++) {
        node = node.next;
    }
    
    node.next = node.next.next;
    
    return head;
};

File: linked lists/19. Remove Nth Node From End of List/removeNthFromEnd.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        length = 0
        
        node = head
        while node:
            node = node.next
            length += 1
        
        # Deleting the head node
        if n == length:
            return head.next
                
        node = head
        # Stop before the n'th node to delete
        for i in range(length - 1 - n):
            node = node.next
        
        # Skip the next node
        node.next = node.next.next
        
        return head